src/rt/aaA.d

```diff
@@ -1013,3 +1023,26 @@ unittest
     assert(typeid(a).getHash(&a) == typeid(a).getHash(&a));
     assert(typeid(a).getHash(&a) == typeid(a).getHash(&a2));
 }
+
+// test AA.reserve
+unittest
+{
+    int[int] aa;
+    assert(aa is null);
+
+    aa.reserve(2000);
+    assert(aa !is null);
+
+    foreach(i;0..2000) aa[i] = i;
+    assert(aa[1337] == 1337);
+
+    // stress test
+    aa.clear();
+    aa.reserve(20_000_000);
+
+    foreach(i;0..32_000) aa[i] = i;
+    assert(aa[31_133] == 31_133);
+
+    foreach(i;0..20_000_000) aa[i] = i;
+    assert(aa[19_999_133] == 19_999_133);
+}
```

Comment:
There should be a test that reserves less than the current length (which will probably fail with the current implementation).